File: app/src/main/java/me/jdvp/baseandroidapplication/application/ViewsModule.java
package me.jdvp.baseandroidapplication.application;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;
import me.jdvp.baseandroidapplication.activity.MainActivity;

/**
 * Module that defines which views will be injected
 *
 * Created by jdvp on 6/18/18.
 */
@Module
abstract class ViewsModule {
    @ContributesAndroidInjector
    abstract MainActivity mainActivity();
}

File: app/src/main/java/me/jdvp/baseandroidapplication/application/ViewsModule.kt
package me.jdvp.baseandroidapplication.application

import dagger.Module
import dagger.android.ContributesAndroidInjector
import me.jdvp.baseandroidapplication.activity.MainActivity
import me.jdvp.baseandroidapplication.fragment.MainFragment

/**
 * Module that defines which views will be injected
 */
@Module
internal abstract class ViewsModule {
    @ContributesAndroidInjector
    abstract fun mainActivity(): MainActivity

    @ContributesAndroidInjector
    abstract fun mainFragment(): MainFragment
}